readme.md

```diff
@@ -8,44 +8,53 @@ This repo contains the [Weaviate](https://weaviate.io) website, it's built using
 
 - Software Dependencies
   - [Ruby](https://www.ruby-lang.org/en/documentation/installation/)
-  - Ruby Development Package (aka ruby-dev)
 
 - Installing Dependencies in **Ubuntu/Debian**
   ```bash
-  # update repositories & install basic build dependencies
-  sudo apt update && sudo apt install -y build-essential bash git rsync
-
-  # update repositories & install rbenv
-  sudo apt update && sudo apt install rbenv ruby-build ruby_dev
-
-  # initialize rbenv
-  rbenv init
-
-  # updates ruby-build local packages
-  git clone https://github.com/rbenv/ruby-build.git "$(rbenv root)"/plugins/ruby-build
-
-  # install ruby 2.7.5
-  rbenv install 2.7.5
-
-  # initialize rbenv
-  export PATH="$HOME/.rbenv/bin:$PATH"
+  # update
+  sudo apt update
+  
+  # install dependencies for installing ruby
+  sudo apt install git curl libssl-dev libreadline-dev zlib1g-dev autoconf bison build-essential libyaml-dev libreadline-dev libncurses5-dev libffi-dev libgdbm-dev
+  
+  # install rbenv[ruby version manager] from its official repo
+  curl -fsSL https://github.com/rbenv/rbenv-installer/raw/HEAD/bin/rbenv-installer | bash
+  
+  # add ~/.rbenv/bin to your $PATH so that you can use the rbenv command line utility
+  echo 'export PATH="$HOME/.rbenv/bin:$PATH"' >> ~/.bashrc
 
-  # set the ruby version locally to 2.7.5
-  rbenv local 2.7.5
-  ```
-  - For installing RubyGems, read [docs](https://jekyllrb.com/docs/installation/ubuntu/)
-  ```bash
-  echo '# Install Ruby Gems to ~/gems' >> ~/.bashrc
-  echo 'export GEM_HOME="$HOME/gems"' >> ~/.bashrc
-  echo 'export PATH="$HOME/gems/bin:$PATH"' >> ~/.bashrc
+  # make rbenv load automatically by adding it to .bashrc
+  echo 'eval "$(rbenv init -)"' >> ~/.bashrc
+  
+  # apply changes to current shell session
   source ~/.bashrc
-  ```
+  
+  # verify if rbenv is installed[Output - rbenv is a function]
+  type rbenv
 
-- Installing Dependencies Windows
+  # install ruby-build plugin from official github repo, adds the rbenv install command
+  git clone https://github.com/rbenv/ruby-build.git
+  
+  # Create this temporary environment variable and run the script[requires sudo]
+  PREFIX=/usr/local sudo ./ruby-build/install.sh
+  
+  # list out ruby versions available for install [choose any version > 2.5.0]
+  rbenv install -l
+  
+  # install a version from the list
+  rbenv install [VERSION_SELECTED]
```

Comment:
Wouldn't giving options to users make this installation more confusing? Giving them a fixed option would make installation easier, that's what I think.

Reply:
There are certain, systems on which certain ruby version is not supported. For instance the already Fixed version `2.7.X` was not supported on mine. 

Had to list down all the version and then install the one which was > 2.5.0

Reply:
But I think, this site needs Ruby version above 2.7.0. This version is also stated in Travis CI's build process. I think lower version of Ruby might break down the Jekyll site.

Webbrick dependency needs Ruby > 2.5.0